### PR TITLE
Run3-gex120 Take care of deprecated codes in RecoParticleFlow/PFSimProducer

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Take care of deprecated codes in RecoParticleFlow/PFSimProducer

#### PR validation:

Use the runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by clacaputo** [comment]
+reconstruction

- technical PR
- no reco changes expected or observed

**Comment by clacaputo** [comment]
+reconstruction

- technical PR
- no reco changes expected or observed

File: RecoParticleFlow/PFSimProducer/plugins/TauHadronDecayFilter.cc [hunk 1]
```diff
@@ -107,9 +110,8 @@ bool TauHadronDecayFilter::filter(edm::Event& iEvent, const edm::EventSetup& iSe
 
 void TauHadronDecayFilter::beginRun(const edm::Run& run, const edm::EventSetup& es) {
   // init Particle data table (from Pythia)
-  edm::ESHandle<HepPDT::ParticleDataTable> pdt;
+  const edm::ESHandle<HepPDT::ParticleDataTable>& pdt = es.getHandle(tokPdt_);
   // edm::ESHandle < DefaultConfig::ParticleDataTable > pdt;
 
-  es.getData(pdt);
-  mySimEvent->initializePdt(&(*pdt));
+  mySimEvent->initializePdt(pdt.product());
```
Comment (perrotta) [inline RecoParticleFlow/PFSimProducer/plugins/TauHadronDecayFilter.cc:116]:
```suggestion
  mySimEvent->initializePdt(&es.getData(tokPdt_));
```

File: RecoParticleFlow/PFSimProducer/plugins/TauHadronDecayFilter.cc [hunk 2]
```diff
@@ -13,7 +13,7 @@
 //
 //
 
-#include "FWCore/Framework/interface/EDFilter.h"
+#include "FWCore/Framework/interface/one/EDFilter.h"
 #include "FWCore/Framework/interface/ESHandle.h"
```
Comment (perrotta) [inline RecoParticleFlow/PFSimProducer/plugins/TauHadronDecayFilter.cc:17]:
```suggestion
```

**Review by perrotta — COMMENTED** [review]
Since you are cleaning, perhaps you can further simplify as follows:

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by clacaputo** [comment]
+reconstruction

- re-sign https://github.com/cms-sw/cmssw/pull/37199#issuecomment-1067835505

**Comment by perrotta** [comment]
+1

**Pull request merged**